presto-hive/src/main/java/com/facebook/presto/hive/HiveOutputTableHandle.java

```diff
@@ -52,6 +55,7 @@ public HiveOutputTableHandle(
 
         this.partitionedBy = ImmutableList.copyOf(requireNonNull(partitionedBy, "partitionedBy is null"));
         this.tableOwner = requireNonNull(tableOwner, "tableOwner is null");
+        this.retentionDays = retentionDays;
```

Comment:
Check for null (the `Optional` reference should never be null, even though what it contains might be)